src/token_generator.js

```diff
@@ -11,23 +11,27 @@ const SyncGrant = AccessToken.SyncGrant;
 
 /**
  * Generate an Access Token for an application user - it generates a random
- * username for the client requesting a token, and takes a device ID as a query
- * parameter.
+ * username for the client requesting a token or generates a token with an
+ * identity if one is provided.
  *
  * @return {Object}
  *         {Object.identity} String random indentity
  *         {Object.token} String token generated
  */
-function tokenGenerator() {
+function tokenGenerator(identity = 0) {
   // Create an access token which we will sign and return to the client
   const token = new AccessToken(
     config.TWILIO_ACCOUNT_SID,
     config.TWILIO_API_KEY,
     config.TWILIO_API_SECRET
   );
 
-  // Assign the generated identity to the token
-  token.identity = nameGenerator();
+  if (identity) {
+    token.identity = identity;
+  } else {
+    // Assign the generated identity to the token
+    token.identity = nameGenerator();
```

Comment:
Perhaps this can be more concise like so:

```js
// Assign the given identity or generate a new one
token.identity = identity || nameGenerator();
```